src/Compilers/CSharp/Test/Emit2/Emit/RuntimeProbing/ModuleCancellationTests.cs

```diff
@@ -0,0 +1,2219 @@
+                void G(long a, CancellationToken token) => throw null;
+                void G(CancellationToken token) => throw null;
+                void G(CancellationToken token, int a) => throw null;
+                void G(int a, int b, CancellationToken token) => throw null;
+                void G<T>(int a, CancellationToken token) => throw null;
+            }
+            """;
+
+        var verifier = CompileAndVerify(source);
+
+        AssertNotInstrumentedWithTokenLoad(verifier, "C.F");
+    }
+
+    [Fact]
+    public void ArgumentReplacement_Overload_ReturnType()
+    {
+        var source = """
+            using System.Threading;
+            using System.Threading.Tasks;
+            
+            class C
+            {
+                void F(CancellationToken token)
+                {
+                    G(1);
+                }
+
+                int G(int a) => throw null;
+                ref int G(int a, CancellationToken token) => throw null;
+            }
+            """;
+
+        var verifier = CompileAndVerify(source);
+
+        AssertNotInstrumentedWithTokenLoad(verifier, "C.F");
+    }
+
+    [Fact]
+    public void ArgumentReplacement_Overload_InstanceToStatic()
+    {
+        var source = """
+            using System.Threading;
+            using System.Threading.Tasks;
+            
+            class C
+            {
+                void F(CancellationToken token)
+                {
+                    G(1);
+                }
+
+                void G(int a)
+                {
+                }
+            
+                static void G(int a, CancellationToken token) => throw null;
+            }
+            """;
+
+        var verifier = CompileAndVerify(source);
+
+        AssertNotInstrumentedWithTokenLoad(verifier, "C.F");
+    }
+
+    [Fact]
+    public void ArgumentReplacement_Overload_StaticToInstance()
+    {
+        var source = """
+            using System.Threading;
+            using System.Threading.Tasks;
+            
+            class C
+            {
+                void F(CancellationToken token)
+                {
+                    G(1);
+                }
+
+                static void G(int a)
+                {
+                }
+            
+                void G(int a, CancellationToken token) => throw null;
+            }
+            """;
+
+        var verifier = CompileAndVerify(source);
+
+        AssertNotInstrumentedWithTokenLoad(verifier, "C.F");
+    }
+
+    [Theory]
+    [InlineData("private")]
+    [InlineData("protected")]
+    [InlineData("internal")]
+    [InlineData("private protected")]
+    [InlineData("internal protected")]
+    public void ArgumentReplacement_Overload_NonMatchingVisibility(string modifier)
+    {
+        var source = $$"""
+            using System.Threading;
+            using System.Threading.Tasks;
+            
+            class B
+            {
+                public static void G(int a) => throw null;
```

Comment:
>public [](http://example.com/codeflow?start=16&length=6)

It would be good to cover other visibility flavors. Consider testing full matrix asserting both success and non-success expectations